realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -220,23 +236,12 @@ void RealSenseNodeFactory::getDevice(rs2::device_list list)
 
 void RealSenseNodeFactory::change_device_callback(rs2::event_information& info)
 {
-	if (info.was_removed(_device))
+	if (_initialized)
```

Comment:
Why not avoid the extra _initialized parameter and use if(_device) instead? See comment for line 397